packages/message/tests/test-DeadLetterMessage.js

```diff
@@ -0,0 +1,114 @@
+'use strict';
+
+const test = require('ava');
+
+const { randomString } = require('@cumulus/common/test-utils');
+
+const { unwrapDeadLetterCumulusMessage } = require('../DeadLetterMessage');
+
+test('unwrapDeadLetterCumulusMessage unwraps an SQS message', async (t) => {
+  const cumulusMessage = {
+    cumulus_meta: {
+      execution_name: randomString(),
+    },
+  };
+  const testMessage = {
+    eventSource: 'aws:sqs',
+    body: JSON.stringify(cumulusMessage),
+  };
+  t.deepEqual(await unwrapDeadLetterCumulusMessage(testMessage), cumulusMessage);
+});
+
+test('unwrapDeadLetterCumulusMessage unwraps an AWS States message', async (t) => {
+  const cumulusMessage = {
+    cumulus_meta: {
+      execution_name: randomString(),
+    },
+  };
+  const testMessage = {
+    source: 'aws.states',
+    detail: {
+      stopDate: Date.now(),
+      output: JSON.stringify(cumulusMessage),
+      status: 'SUCCEEDED',
+    },
+  };
+  const actual = await unwrapDeadLetterCumulusMessage(testMessage);
+  const expected = {
+    ...cumulusMessage,
+    meta: { status: 'completed' },
+  };
+  expected.cumulus_meta.workflow_stop_time = testMessage.detail.stopDate;
+  t.deepEqual(actual, expected);
+});
+
+test('unwrapDeadLetterCumulusMessage unwraps an AWS States message with only input', async (t) => {
+  const cumulusMessage = {
+    cumulus_meta: {
+      execution_name: randomString(),
+    },
+  };
+  const testMessage = {
+    source: 'aws.states',
+    detail: {
+      stopDate: Date.now(),
+      input: JSON.stringify(cumulusMessage),
+      status: 'RUNNING',
+    },
+  };
+
+  const actual = await unwrapDeadLetterCumulusMessage(testMessage);
+  const expected = {
+    ...cumulusMessage,
+    meta: { status: 'running' },
+  };
+  expected.cumulus_meta.workflow_stop_time = testMessage.detail.stopDate;
+  t.deepEqual(actual, expected);
+});
+
+test('unwrapDeadLetterCumulusMessage unwraps an AWS states message within an SQS record', async (t) => {
+  const stopDate = Date.now();
+  const cumulusMessage = {
+    cumulus_meta: {
+      execution_name: randomString(),
+    },
+  };
+
+  const testStatesMessage = {
+    source: 'aws.states',
+    detail: {
+      stopDate,
+      output: JSON.stringify(cumulusMessage),
+      status: 'SUCCEEDED',
+    },
+  };
+  const testSqsMessage = {
+    sourceEvent: 'aws:sqs',
+    body: JSON.stringify(testStatesMessage),
+  };
+
+  const actual = await unwrapDeadLetterCumulusMessage(testSqsMessage);
+  const expected = {
+    ...cumulusMessage,
+    meta: { status: 'completed' },
+  };
+  expected.cumulus_meta.workflow_stop_time = stopDate;
+  t.deepEqual(actual, expected);
+});
+
+test('unwrapDeadLetterCumulusMessage returns wrapped message on error', async (t) => {
+  const invalidMessage = {
+    Body: 'Not a json object',
+  };
+  t.deepEqual(await unwrapDeadLetterCumulusMessage(invalidMessage), invalidMessage);
+});
+
+test('unwrapDeadLetterCumulusMessage returns an non-unwrappable message', async (t) => {
```

Comment:
I could be wrong - but do we need an additional unit test to catch whatever buggy behavior it is that PODAAC was running into?

Reply:
No, the problem was that in the original implementation the `write-db-dlq-records-to-s3` used the `unwrapDeadLetterCumulusMessage` in it's lambda *to get the execution name*, but *not* to write the message.   

There wasn't anything wrong with the unwrap logic, although it's pretty arguable if we should *only* write the Cumulus message.    This PR only moved it to the message package, the OG units should be fine as is, aside the updates due to the call to `getCumulusMessageFromExecutionEvent`